examples/coordination/README.md

```diff
@@ -0,0 +1,156 @@
+used.
+
+```go
+for {
+    session, err := db.Coordination().OpenSession(ctx, path)
+    if err != nil {
+		// The context is canceled.
+        break		
+    }
+
+    lease, err := session.AcquireSemaphore(ctx, semaphore, coordination.Exclusive, options.WithEphemeral(true))
+    if err != nil {
+        session.Close(ctx)
+        continue
+    }
+
+    // The lock is acquired. 
+    go doWork(lease.Context())
+	
+    // Wait until the lock is released.
+    <-lease.Context().Done():
+	
+    // The lock is not acquired anymore.
+    cancelWork()
+}
+```
+
+## Workers
+
+This is an example of distributing long-running tasks among multiple workers. Consider there is a number of tasks that 
+need to be processed simultaneously by a pool of workers. Each task is processed independently and the order of
+processing does not matter. A worker has a fixed capacity that defines how many tasks it is ready to process at a time.
+
+Any single task is associated with a Coordination Service semaphore. When the application starts, it grabs all task 
+semaphores in order to acquire at least `capacity` of them. When it happens, it releases excessive ones and wait until 
+the application finishes. Until then, it starts a new task for every acquired semaphore and waits until the number of 
+running tasks becomes equal to the capacity of the worker.
+
+To start the application with the [YDB Docker database instance](https://ydb.tech/en/docs/getting_started/self_hosted/ydb_docker) run
+
+```bash
+$ go build 
+$ YDB_ANONYMOUS_CREDENTIALS=1 ./lock -ydb grpc://localhost:2136/local -path /local/test --semaphore-prefix job- --tasks 10 --capacity 4
+```
+
+This example uses ephemeral semaphores which are always acquired exclusive. However, in a real application you may 
+want to use persistent semaphores in order to store the state of workers in attached data. The following pseudocode 
+shows how it is used.
+
+```go
+for {
+    session, err := db.Coordination().OpenSession(ctx, path)
+    if err != nil {
+        // The context is canceled.
+        break
+    }
+
+    semaphoreCtx, semaphoreCancel := context.WithCancel(ctx)
+    capacitySemaphore := semaphore.NewWeighted(capacity)
+    leaseChan := make(chan *coordination.Lease)
+    for _, name := range tasks {
+        go awaitSemaphore(semaphoreCtx, semaphoreCancel, session, name, capacitySemaphore, leaseChan)
+    }
+
+    tasksStarted := 0
+loop:
+    for {
+        lease := <-leaseChan
+			
+        // Run a new task every time we acquire a semaphore.
+        go doWork(lease.Context())
+        
+        tasksStarted++
+        if tasksStarted == capacity {
+            break
+        }
+    }
+
+    // The capacity is full, cancel all Acquire operations.
+    semaphoreCancel()
+	
+    // Wait until the session is alive.
+    <-session.Context().Done()
+	
+    // The tasks must be stopped since we do not own the semaphores anymore.
+    cancelTasks()
```

Comment:
Why we need explicit stop tasks?

we call doWork with lease.Context() and the context will be cancelled when session will be closed (session context cancelled)

Reply:
Good call! I'll change this to 'wait for the tasks to complete'